src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,49 @@
 package core.basesyntax;
 
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Date;
+
 public class SalaryInfo {
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+
+        ArrayList<String> dataList = new ArrayList<>(Arrays.asList(data));
+        SimpleDateFormat formatter = new SimpleDateFormat("dd.MM.y");
+
+        try {
+            Date dateFrom1 = formatter.parse(dateFrom);
+            Date dateTo1 = formatter.parse(dateTo);
+
+            for (int i = 0; i < dataList.size(); i++) {
+                if (dateFrom1.after(formatter.parse(dataList.get(i).substring(0,
```

Comment:
I think it's better to use split(" ") here and get element with index `0` to get date